supabase/migrations/20251128094912_optimise_supabase.sql

```diff
@@ -0,0 +1,51 @@
+-- ============================================================================
+-- MIGRATION: Optimize search_bookmarks_debugging with similarity scoring
+-- Created: 2025-11-28
+-- Purpose: Enhance search function with URL matching and weighted similarity ranking
+-- ============================================================================
+
+set check_function_bodies = off;
+
+CREATE OR REPLACE FUNCTION public.search_bookmarks_debugging(search_text character varying)
+ RETURNS TABLE(id bigint, user_id uuid, inserted_at timestamp with time zone, title extensions.citext, url text, description text, ogimage text, screenshot text, category_id bigint, trash boolean, type text, meta_data jsonb, sort_index text)
+ LANGUAGE plpgsql
+ SET search_path TO 'public', 'extensions'
+AS $function$BEGIN
+    PERFORM set_limit(0.6);
+
+    RETURN QUERY
+    SELECT b.*
+    FROM everything b
+    WHERE
+        (search_text % ANY(STRING_TO_ARRAY(b.title || b.description, ' ')))
+        OR b.url ILIKE '%' || search_text || '%'
+        OR EXISTS (
+            SELECT 1
+            FROM jsonb_each_text(b.meta_data) x(key, value)
+            WHERE key IN ('img_caption', 'ocr')
+              AND value ILIKE '%' || search_text || '%'
+        )
+    ORDER BY
+        (
+            similarity(b.url, search_text) * 0.6 +
+            similarity(b.title, search_text) * 0.5 +
+            similarity(b.description, search_text) * 0.3 +
+            similarity(COALESCE(b.meta_data->>'ocr',''), search_text) * 0.1 +
+            similarity(COALESCE(b.meta_data->>'img_caption',''), search_text) * 0.15
```

Comment:
@coderabbitai Mathematically this works for relative ordering, but conventionally weights should sum to 100%
  for clarity. Consider normalizing.

Should we normalize it?

Reply:
@B-a-l-aj-i Please test this change.